packages/core/src/sdk/deliveryPromise/useDeliveryPromise.ts

```diff
@@ -435,6 +441,84 @@ export function useDeliveryPromise({
     []
   )
 
+  function getBadgeLabel(value: DeliveryType, isAvailable: boolean) {
+    const labelMap: Record<
+      DeliveryType,
+      { available: string; unavailable: string }
+    > = {
+      delivery: {
+        available:
+          deliveryPromiseSettings?.deliveryPromiseBadges?.delivery ??
+          'Available for shipping',
+        unavailable:
+          deliveryPromiseSettings?.deliveryPromiseBadges?.deliveryUnavailable ??
+          'Unavailable for shipping',
+      },
+      'pickup-in-point': {
+        available:
+          deliveryPromiseSettings?.deliveryPromiseBadges?.pickupInPoint ??
+          'Available for pickup',
+        unavailable:
+          deliveryPromiseSettings?.deliveryPromiseBadges
+            ?.pickupInPointUnavailable ?? 'Unavailable for pickup',
+      },
+    }
+
+    return labelMap[value]
+      ? isAvailable
+        ? labelMap[value].available
+        : labelMap[value].unavailable
+      : value
+  }
+
+  function getDeliveryPromiseBadges() {
+    if (!deliveryPromiseBadges) return []
```

Comment:
todo: when deliveryPromiseBadges is an empty array, it's a truthy value, so we have to also check if the length is 0
```suggestion
    // Only add unavailable badges if at least one delivery method is available
    if (!deliveryPromiseBadges || deliveryPromiseBadges.length === 0) return []
```